### PR TITLE
Buildkite migration is now complete, removing fleet-server jobs

### PR DESCRIPTION
**Pull request by jlind23**
## What is the problem this PR solves?

// Please do not just reference an issue. Explain WHAT the problem this PR solves here.

## How does this PR solve the problem?

// Explain HOW you solved the problem in your code. It is possible that during PR reviews this changes and then this section should be updated.

## Design Checklist

- [ ] I have ensured my design is stateless and will work when multiple fleet-server instances are behind a load balancer.
- [ ] I have or intend to scale test my changes, ensuring it will work reliably with 100K+ agents connected.
- [ ] I have included fail safe mechanisms to limit the load on fleet-server: rate limiting, circuit breakers, caching, load shedding, etc.

## Checklist

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/fleet-server#changelog)

### CODE REVIEW

**Comment by michel-laterman** [comment]
Looks like `.ci/scripts/test-release.sh` is used by buildkite, it should probably be moved into .buildkite/scripts

File: Makefile [hunk 1]
```diff
@@ -154,7 +154,7 @@ test: prepare-test-context  ## - Run all tests
 
 .PHONY: test-release
 test-release:  ## - Check that all release binaries are created
-	./.ci/scripts/test-release.sh $(DEFAULT_VERSION)
+	./.buildkite/scripts/test-release.sh $(DEFAULT_VERSION)
```
Comment (juliaElastic) [inline Makefile:157]: Should the `test-release.sh` file be moved to `.buildkite`?

Comment (jlind23) [reply Makefile:157]: Yes 🤦🏼

**Review by juliaElastic — APPROVED** [review]

**Comment by jlind23** [comment]
the buildkite build will fail until the unified release for 8.12 is done.

**Comment by jlind23** [comment]
buildkite test this

**Comment by jlind23** [comment]
buildkite test this

**Comment by jlind23** [comment]
Blocked by https://github.com/elastic/fleet-server/pull/2999

**Pull request merged**